src/Configuration/CFGCommon/CFGCommon.cpp

```diff
@@ -316,6 +316,17 @@ uint64_t CFG_convert_string_to_u64(std::string string, bool no_empty,
   return value;
 }
 
+std::string CFG_convert_number_to_unit_string(uint64_t number) {
+  static const std::vector<std::string> units = {"", "K", "M", "G", "T"};
+  size_t unitIndex = 0;
+  while (number >= 1024 && unitIndex < units.size() - 1) {
```

Comment:
Please use the original checking that I gave you.

You must check (% 1024) == 0.

Reply:
If the number is 1500. Then we should not convert it (because the result will be 1k which is wrong) in this case unless we support float number. 

I know you did check (% 1024) == 0 in the extractDeviceList() [by the way it is wrong checking]. But remember this is a generic number, it is not used by flashsize conversion.

Reply:
Ok. Fixed.